From 91ca2d2cc4c053c58b1685c1bd484f42b078817d Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Wed, 19 Jan 2022 15:19:29 +0100
Subject: [PATCH 3/3] Makefile.am: update rpath link

since commit 6339062 Move liblttng-ust to 'src/lib/',
liblttng-ust.so/liblttng-ust-common.so/liblttng-ust-tracepoint.so
's location changed from one dir to multiple dirs. which make below
error:
ld: warning: liblttng-ust-common.so.1, needed by ../../../src/lib/lttng-ust/.libs/liblttng-ust.so, not found (try using -rpath or -rpath-link)
ld: warning: liblttng-ust-tracepoint.so.1, needed by ../../../src/lib/lttng-ust/.libs/liblttng-ust.so, not found (try using -rpath or -rpath-link)

Upstream-Status: Submitted [https://github.com/lttng/lttng-ust/pull/61]

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 doc/examples/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/examples/Makefile.am b/doc/examples/Makefile.am
index 8ee0564..e2ccef9 100644
--- a/doc/examples/Makefile.am
+++ b/doc/examples/Makefile.am
@@ -142,7 +142,7 @@ all-local:
 				CFLAGS='$(CFLAGS)' \
 				AM_CFLAGS='$(AM_CFLAGS)' \
 				LDFLAGS="$(LDFLAGS)" \
-				AM_LDFLAGS='$(AM_LDFLAGS) -L../../../src/lib/lttng-ust/.libs -Wl,-rpath="$(PWD)/../../src/lib/lttng-ust/.libs/" -Wl,-rpath-link="$(PWD)/../../src/lib/lttng-ust/.libs/"' \
+				AM_LDFLAGS='$(AM_LDFLAGS) -L../../../src/lib/lttng-ust/.libs -Wl,-rpath="$(PWD)/../../src/lib/lttng-ust/.libs/" -Wl,-rpath-link="$(PWD)/../../src/lib/lttng-ust/.libs/:$(PWD)/../../src/lib/lttng-ust-tracepoint/.libs:$(PWD)/../../src/lib/lttng-ust-common/.libs/"' \
 				LTTNG_GEN_TP_PATH="$$rel_src_subdir$(top_srcdir)/tools/" \
 				AM_V_P="$(AM_V_P)" \
 				AM_V_at="$(AM_V_at)" \
-- 
2.17.1

